access/packages/api/src/ucanto/capabilities.js

```diff
@@ -0,0 +1,73 @@
+import { capability, Failure, URI } from '@ucanto/validator'
+// eslint-disable-next-line no-unused-vars
+import * as Types from '@ucanto/interface'
+
+/**
+ * Check URI can be delegated
+ *
+ * @param {string} child
+ * @param {string} parent
+ */
+const canDelegateURI = (child, parent) => {
+  if (parent.endsWith('*')) {
+    return child.startsWith(parent.slice(0, -1))
+      ? true
+      : new Failure(`${child} does not match ${parent}`)
+  }
+
+  return child === parent
+    ? true
+    : new Failure(`${child} is different from ${parent}`)
+}
+
+/**
+ * Checks that `with` on claimed capability is the same as `with`
+ * in delegated capability. Note this will ignore `can` field.
+ *
+ * @param {Types.ParsedCapability} child
+ * @param {Types.ParsedCapability} parent
+ */
+function equalWith(child, parent) {
+  return (
+    child.with === parent.with ||
+    new Failure(
+      `Can not derive ${child.can} with ${child.with} from ${parent.with}`
+    )
+  )
+}
+
+export const identityValidate = capability({
+  can: 'identity/validate',
+  with: URI.match({ protocol: 'did:' }),
+  caveats: {
+    as: URI.string({ protocol: 'mailto:' }),
+  },
+  derives: (child, parent) => {
+    return (
+      canDelegateURI(child.caveats.as, parent.caveats.as) &&
+      equalWith(child, parent)
+    )
+  },
+})
+
+export const identityRegister = capability({
+  can: 'identity/register',
+  with: URI.match({ protocol: 'mailto:' }),
+  caveats: {
+    as: URI.string({ protocol: 'did:' }),
+  },
+  derives: (child, parent) =>
+    canDelegateURI(child.caveats.as, parent.caveats.as) &&
+    canDelegateURI(child.with, parent.with),
+})
+
+/**
+ * `identity/identify` can be derived from any of the `store/*`
+ * capability that has matichng `with`. This allows store service
+ * to identify account based on any user request.
+ */
+export const identityIdentify = capability({
```

Comment:
As discussed before this really should be something along the lines of 

https://github.com/web3-storage/ucanto/blob/c73a31ac122af1319183bf3be6447d70e01d6c55/w3/store/src/identity/capability.js#L72-L84

Otherwise "store/*" service provider will not be able to invoke this capability. Alternatively we could delegate "identity/identify" capability from this service to "store/*" service but that is IMO not a better option because:

1. Instead of tracking dependencies in source code and with packages we'd end up doing it through brittle configurations.
2. Connection between "store/*" request and "identity/identify" that it caused will be lost.
3. Delegating unconditional "identity/identify" capability to "store/*" service provider would imply that compromise in later could allow attacker to obtain identity info about all of our accounts. Deriving capability instead would not have that problem as identity could only be resolved for upload target DIDs.

I agree that it is unfortunate that this implies that `identity/identify` has to depend on defined capabilities for `store/*`, but then again this dependency would exist even if we leave it out of code because "identity/*" service needs to know about:

1. Which capabilities it can be derived from.
2. How capabilities it can be derived from can be compared & validated.

Which is precisely what capability definitions do. I think we just need to factor out all the related capabilities into separate module and import them from `store/*` service provider, client and `identity` service provider and client.

Reply:
i really dont like this, we should figure out a better option. What about other services that need to call identity/identify but dont deal with store/* ?

Reply:
The thing is services needs to know about each other when they interact in order, there’s no way around this. Now you can thread this knowledge either:

1. Delegate capabilities for interaction (from provider to caller)
2. Define capabilities in provider so caller can just relay user requests.
3. Require that user provide additional capabilities so that service can relay those.

In the end you need to encode that knowledge in one of the three interacting actors and regardless of the choice there are tradeoffs & added complexity.

That said maybe rethinking accounts as we discussed is going to help a bit, specifically linking email increasing account capacity would remove dependency here.
Yet it will create dependency elsewhere vitrification & capacity increase is still going to be distinct capabilities that will either:

1. Be derived from one another
2. Be unlinked from one another

I would still argue that deriving “account/activate” from “identity
/verify” is account services business logic and therefor it makes sense to encode that in its code. It also means that in the future this logic could change without necessarily coordinating with other actors.

If you were to go other way round and just delegate “account/activate” to verifier changing business logic of account service would become more difficult & would require coordinated changes in both services